libraries/scripting/dependencies/src/kotlin/script/experimental/dependencies/ExternalDependenciesResolver.kt

```diff
@@ -7,16 +7,30 @@ package kotlin.script.experimental.dependencies
 
 import java.io.File
 import kotlin.script.experimental.api.ResultWithDiagnostics
+import kotlin.script.experimental.api.SourceCode
 import kotlin.script.experimental.api.valueOrNull
 
 open class RepositoryCoordinates(val string: String)
 
 interface ExternalDependenciesResolver {
+    interface Options {
+        interface Parser {
```

Comment:
I think that we do not need to expose the flexibility of parsing here. This simply gives an additional coupling between the resolver and annotations, which is not good to my mind.
It seems to me that if user wants to have its own way to parse options, he should rather create a new annotation and use his own helpers that resolve for the annotation, and put the parser he wants there. 
And in our own helpers, we'd better stick to a single consistent (albeit simple) syntax for the options.
This means that the parser interface and parameters should be dropped, the default parser put into the `impl` as an implementation detail and used internally directly.

Reply:
Sounds good ✅